contracts/core/Allo.sol

```diff
@@ -33,57 +57,108 @@ contract Allo is Initializable {
     // takes fee from user
     // validates that the owner is actually allowed to use the identity
     function createPool(
-        PoolData memory /*_poolData*/
+        bytes32 _identityId,
+        address _allocationStrategy,
+        address _distributionStrategy,
+        Metadata _metadata
     ) external pure returns (uint) {
-        uint32 _poolId = 0;
 
-        // todo: return the poolId? what do we want to return here?
-        return _poolId;
+        if (!registry.isMemberOfIdentity(_identityId, msg.sender)) {
+            revert NO_ACCESS_TO_ROLE();
+        }
+        
+        // ASK should we clone _allocationStrategy / _distributionStrategy or just use them directly?
+        // bytes32 salt = keccak256(abi.encodePacked(msg.sender, nonce, "allocationStrategy"));
+        // address clone = ClonesUpgradeable.cloneDeterministic(_allocationStrategy, salt);
+        
```

Comment:
@codenamejason @zobront @KurtMerbeth

Reply:
For the allo-strategies we should create clones, since it's cheaper.

For custom strategies I would suggest the same, since they will be re-usable. But I'm also open for direct use

Reply:
Creating clones does seem nicer <3
Okie we'll pair on this on Monday and get it sorted